src/Combinatorics/SimplicialComplexes.jl

```diff
@@ -0,0 +1,351 @@
+    SimplicialComplex,
+    betti_numbers,
+    dim,
+    euler_characteristic,
+    facets,
+    f_vector,
+    h_vector,
+    minimal_nonfaces,
+    nvertices,
+    vertexindices,
+    stanley_reisner_ideal,
+    stanley_reisner_ring,
+    load_simplicialcomplex,
+    save_simplicialcomplex,
+    complex_projective_plane,
+    real_projective_plane,
+    klein_bottle,
+    torus # requires a distinction from, e.g., an algebraic group
+
+################################################################################
+##  Constructing
+################################################################################
+
+struct SimplicialComplex
+    pm_simplicialcomplex::Polymake.BigObject
+end
+
+pm_object(K::SimplicialComplex) = K.pm_simplicialcomplex
+
+
+@doc Markdown.doc"""
+    SimplicialComplex(generators::Union{Vector{Vector{Int}}, Vector{Set{Int}}})
+
+Construct an abstract simplicial complex from a set of faces.
+While arbitrary nonnegative integers are allowed as vertices, they will be relabeled to consecutive integers starting at 1.
+
+# Example
+```jldoctest
+julia> K = SimplicialComplex([[1,2,3],[2,3,4]])
+Abstract simplicial complex of dimension 2 on 4 vertices
+```
+# Simplicial complex comprising the empty set only
+```jldoctest
+julia> empty = SimplicialComplex(Vector{Set{Int}}([]))
+Abstract simplicial complex of dimension -1 on 0 vertices
+```
+# Example with relabeling
+The original vertices can be recovered.
+```jldoctest
+julia> L = SimplicialComplex([[0,2,17],[2,17,90]]);
+
+julia> facets(L)
+2-element Vector{Vector{Int64}}:
+ [1, 3, 2]
+ [3, 4, 2]
+
+julia> vertexindices(L)
+4-element Vector{Int64}:
+  0
+  2
+ 17
+ 90
+```
+"""
+function SimplicialComplex(generators::Union{Vector{Vector{Int}}, Vector{Set{Int}}})
+    K = Polymake.topaz.SimplicialComplex(INPUT_FACES=generators)
+    SimplicialComplex(K)
+end
+
+################################################################################
+##  Auxiliary
+################################################################################
+
+function _vertexindices(K::Polymake.BigObject)
+    if Polymake.exists(K,"VERTEX_INDICES")
+        return Vector{Int}(K.VERTEX_INDICES)
+    else
+        return Vector{Int}(1:K.N_VERTICES)
+    end
+end
+
+vertexindices(L::SimplicialComplex) = _vertexindices(pm_object(L))
+
+# currently unused
+_reindexset(M::Set{Int}, ind::Vector{Int}) = [ ind[x+1] for x in M ]
+
+function _characteristic_vector(M::Set{Int}, n::Int)
+    chi = zeros(Int, n)
+    for x in M
+        chi[x] = 1
+    end
+    return chi
+end
+
+################################################################################
+##  Properties
+################################################################################
+
+@doc Markdown.doc"""
+    nvertices(K::SimplicialComplex)
+
+Number of vertices of the abstract simplicial complex `K`.
+
+# Example
+```jldoctest
+julia> nvertices(torus())
+7
+```
+"""
+nvertices(K::SimplicialComplex)::Int = pm_object(K).N_VERTICES
```

Comment:
This inserts a *conversion* of the function result to `Int`, but I am guessing that `N_VERTICES` already is an `Int`, so a type assertion would be better (as it avoids inserting a call to `convert(Int, ...)`):
```suggestion
nvertices(K::SimplicialComplex) = pm_object(K).N_VERTICES::Int
```

The same elsewhere